akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -671,4 +671,20 @@ object Source {
   def unfoldResourceAsync[T, S](create: () ⇒ Future[S], read: (S) ⇒ Future[Option[T]], close: (S) ⇒ Future[Done]): Source[T, NotUsed] =
     Source.fromGraph(new UnfoldResourceSourceAsync(create, read, close))
 
+  /**
+   * Materializes into a sink connected to a source, i.e. the sink pushes into the source:
+   *
+   * +----------+       +----------+
+   * >   Sink   |------>|  Source  >
+   * +----------+       +----------+
+   *
+   * Should be provided by Akka Streams, see https://github.com/akka/akka/issues/24853.
+   */
+  def sinkToSource[M]: RunnableGraph[(Sink[M, NotUsed], Source[M, NotUsed])] =
```

Comment:
We need to ponder the name some more...
I have no good name today in mind though...

Reply:
This is the best I can come up with

Reply:
Regarding the name: This seems like the opposite of [Flow#fromSinkAndSource](https://doc.akka.io/api/akka/current/akka/stream/scaladsl/Flow$.html#fromSinkAndSource[I,O](sink:akka.stream.Graph[akka.stream.SinkShape[I],_],source:akka.stream.Graph[akka.stream.SourceShape[O],_]):akka.stream.scaladsl.Flow[I,O,akka.NotUsed]). It seems like this could live on `Flow` (the class not the object), the idea would be like breaking an existing flow into a sink and source.

So something like:
```
class Flow[-In, +Out, +Mat] {
.
.
.
def toSinkAndSource: (Sink[In],Source[Out]) = 
  Source
    .asSubscriber[In]
    .via(this)
    .toMat(Sink.asPublisher[Out](fanout = false))(Keep.both)
    .mapMaterializedValue {
      case (sub, pub) ⇒ (Sink.fromSubscriber(sub), Source.fromPublisher(pub))
    }.run()
```

Of course, you have to `run` it in order to actually get the sink and the source, so that might not be desirable.